airflow/providers/databricks/hooks/databricks.py

```diff
@@ -352,3 +355,44 @@ def get_repo_by_path(self, path: str) -> Optional[str]:
             return str(result['object_id'])
 
         return None
+
+    def import_notebook(self, notebook_name: str, raw_code: str, language: str, overwrite: bool = True, format: str = 'SOURCE'):
+        """
+        Import a local notebook from Airflow into Databricks FS. Notebooks saved to /Shared/airflow dbfs
+
+        Utility function to call the ``2.0/workspace/import`` endpoint.
+
+        :param notebook_name: String name of notebook on Databricks FS
+        :param raw_code: String of non-encoded code
+        :param language: Use one of the following strings 'SCALA', 'PYTHON', 'SQL', OR 'R'
+        :param overwrite: Boolean flag specifying whether to overwrite existing object. It is true by default
+        :return: full dbfs notebook path
+        """
+        #enforce language options
+        language_options = ['SCALA', 'PYTHON', 'SQL', 'R']
+        if language.upper() not in language_options:
+            raise ValueError(f"results: language must be one of the following: {str(language_options)}")
+
+        # enforce format options
+        format_options = ['SOURCE', 'HTML', 'JUPYTER', 'DBC']
+        if format.upper() not in format_options:
+            raise ValueError(f"results: format must be one of the following: {str(format_options)}")
+
+        # encode notebook
+        encodedBytes = base64.b64encode(raw_code.encode("utf-8"))
+        encodedStr = str(encodedBytes, "utf-8")
+
+        # create parent directory if not exists
+        self._do_api_call(WORKSPACE_MKDIR_ENDPOINT, {'path': "/Shared/airflow"})
```

Comment:
Will be more applicable if the user is able to specify a path to create

Reply:
So, I thought this same thing based on Databricks docs, but I've run it and it doesn't fail. Should I still add the catch?

Reply:
Looks like _do_api_call returns response.json, so I would take that value and then return it at the end.  So success, or RESOURCE_ALREADY_EXISTS.  Rather than returning the path that the user already has because they passed it in via parameters.  Addresses @eladkal's return question at the same time.

I think catch was the wrong word. Rather than "catching" the exception and running something else in an `except` block, more what I mean is saving that output and somehow reporting that to the user.

Reply:
Despite what the docs say the `RESOURCE_ALREADY_EXISTS` error doesn't appear when I re-run the task. Nothing gets returned from the api endpoint.

Reply:
`RESOURCE_ALREADY_EXISTS` will be returned if you're trying to create a directory with the same name as existing non-directory object (notebook, for example).